hydra-node/src/Hydra/Chain/CardanoClient.hs

```diff
@@ -19,6 +22,10 @@ data QueryException
   = QueryAcquireException AcquiringFailure
   | QueryEraMismatchException EraMismatch
   | QueryProtocolParamsConversionException ProtocolParametersConversionError
+  | QueryProtocolParamsEraNotSupported Text
+  | QueryProtocolParamsEncodingFailureOnEra Text Text
+  | QueryEraNotInCardanoModeFailure Text
+  | QueryNotShelleyBasedEraException Text
```

Comment:
Could: hold proper types in these constructors instead of `Text`

```suggestion
  | QueryNotShelleyBasedEraException (forall era. CardanoEra era)
```

..should work

Reply:
Used AnyCardanoEra instead for this purpose, as it was easier.